src/fileio.jl

```diff
@@ -65,11 +73,19 @@ function _convert(
 end
 
 # SHA256
-_convert(::Type{DataFormat{:SHA256}}, x; kw...) = bytes2hex(sha256(string(x)))
-function _convert(::Type{DataFormat{:SHA256}}, img::AbstractArray{<:Colorant}; kw...)
+maybe_encode(::Type{DataFormat{:SHA256}}, x; kw...) = _sha256(string(x))
+maybe_encode(::Type{DataFormat{:SHA256}}, x::AbstractString) = _sha256(_ignore_crlf(x))
+maybe_encode(::Type{DataFormat{:SHA256}}, x::AbstractArray{<:AbstractString}) = _sha256(_join(x))
+function maybe_encode(::Type{DataFormat{:SHA256}}, img::AbstractArray{<:Colorant}; kw...)
     # encode image into SHA256
-    size_str = bytes2hex(sha256(reinterpret(UInt8,[map(Int64,size(img))...])))
-    img_str = bytes2hex(sha256(reinterpret(UInt8,vec(rawview(channelview(img))))))
+    size_str = _sha256(reinterpret(UInt8,[map(Int64,size(img))...]))
+    img_str = _sha256(reinterpret(UInt8,vec(rawview(channelview(img)))))
 
     return size_str * img_str
 end
+
+
+# Helpers
+_join(x::AbstractArray{<:AbstractString}) = mapreduce(_ignore_crlf, (x,y)->x*"\n"*y, x)
+_sha256(x) = bytes2hex(sha256(x))
+_ignore_crlf(x::AbstractString) = replace(x, "\r"=>"")
```

Comment:
Should we just say we are ignoring `\r` ?
```suggestion
_ignore_linefeed(x::AbstractString) = replace(x, "\r"=>"")
```

Reply:
I assume you're suggesting Carriage Return (`\r`) instead of linefeed (`\n`)?